From 79c911b922dd4d73a779acb0ae19a65e7294573f Mon Sep 17 00:00:00 2001
From: Raviteja Narayanam <raviteja.narayanam@xilinx.com>
Date: Fri, 13 Mar 2020 11:48:03 +0530
Subject: [PATCH 1270/1852] i2c: xiic: Correct the datatype for rx_watermark

commit 708f6872d06924300caa50c7513342ffc52a4ed5 from
https://github.com/Xilinx/linux-xlnx.git

The message length data type should be u16 as per the i2c_msg structure

Signed-off-by: Raviteja Narayanam <raviteja.narayanam@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/i2c/busses/i2c-xiic.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-xiic.c b/drivers/i2c/busses/i2c-xiic.c
index 6c732df087d7..5031c5c7d770 100644
--- a/drivers/i2c/busses/i2c-xiic.c
+++ b/drivers/i2c/busses/i2c-xiic.c
@@ -649,7 +649,8 @@ static int xiic_busy(struct xiic_i2c *i2c)
 
 static void xiic_start_recv(struct xiic_i2c *i2c)
 {
-	u8 rx_watermark, cr = 0, rfd_set = 0;
+	u16 rx_watermark;
+	u8 cr = 0, rfd_set = 0;
 	struct i2c_msg *msg = i2c->rx_msg = i2c->tx_msg;
 	unsigned long flags;
 
-- 
2.31.1

